evalml/automl/automl_search.py

```diff
@@ -600,6 +600,10 @@ def _find_best_pipeline(self):
                     X_train, X_threshold_tuning, y_train, y_threshold_tuning = split_data(X_train, y_train, self.problem_type,
                                                                                           test_size=0.2,
                                                                                           random_seed=self.random_seed)
+                if hasattr(self.data_splitter, "transform_sample"):
```

Comment:
I'm not a big fan of using `hasattr` here. Ideally, the logic of knowing when and how to resample the data should live in one place. Right now it's in `BaseUnderSamplingSplitter`, here, and soon it will have to go into the [train_pipelines](https://github.com/alteryx/evalml/pull/1913/files#diff-8b4df814812e6f5df24c600bc7f429b036e0e2f78613bfbdcdf0ed3ab86bffd6R912) method that's going in #1913 . 

I don't have a clear alternative yet 😅  so I don't think this is something we need to change now. Maybe the engine can handle this responsibility and we can consider this in #1914

Reply:
@freddyaboulton Right, I agree, it does seem to be in too many places. Potentially if we move training the best pipeline to the engine, we could reduce the number of places this appears in. As for the `hasattr` call, we could also check the subclass name, if that's cleaner/preferred?

Reply:
@bchen1116 I think we can leave it as is for now!